OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/AdminControllerTests.cs

```diff
@@ -0,0 +1,956 @@
+        var changedDirection = new DirectionDto()
+        {
+            Id = 1,
+            Title = "ChangedTitle",
+        };
+        sensitiveDirectionService.Setup(x => x.Update(changedDirection)).ReturnsAsync(changedDirection);
+
+        // Act
+        var result = await controller.UpdateDirections(changedDirection).ConfigureAwait(false) as OkObjectResult;
+
+        // Assert
+        Assert.That(result, Is.Not.Null);
+        Assert.AreEqual(200, result.StatusCode);
+    }
+
+    [Test]
+    public async Task UpdateDirections_WhenModelIsInvalid_ReturnsBadRequestObjectResult()
+    {
+        // Arrange
+        var user = new ClaimsPrincipal(new ClaimsIdentity(
+            new Claim[]
+            {
+                new Claim(ClaimTypes.Role, nameof(Role.TechAdmin).ToLower()),
+            }));
+        controller.ControllerContext.HttpContext.User = user;
+        controller.ModelState.AddModelError("UpdateDirection", "Invalid model state.");
+
+        // Act
+        var result = await controller.UpdateDirections(direction).ConfigureAwait(false);
+
+        // Assert
+        Assert.That(result, Is.TypeOf<BadRequestObjectResult>());
+        Assert.That(((BadRequestObjectResult)result).StatusCode, Is.EqualTo(400));
+    }
+
+    [Test]
+    public async Task UpdateDirections_WhenUserIsNotTechAdmin_ReturnsObjectResult()
+    {
+        // Arrange
+        var user = new ClaimsPrincipal(new ClaimsIdentity(
+            new Claim[]
+            {
+                new Claim(ClaimTypes.Role, nameof(Role.MinistryAdmin).ToLower()),
+            }));
+        controller.ControllerContext.HttpContext.User = user;
+        var changedDirection = new DirectionDto()
+        {
+            Id = 1,
+            Title = "ChangedTitle",
+        };
+        sensitiveDirectionService.Setup(x => x.Update(changedDirection)).ReturnsAsync(changedDirection);
+
+        // Act
+        var result = await controller.UpdateDirections(changedDirection).ConfigureAwait(false);
+
+        // Assert
+        Assert.That(result, Is.InstanceOf<ObjectResult>());
+    }
+
+    [Test]
+    [TestCase(1)]
+    public async Task DeleteDirectionById_WhenIdIsValid_ReturnsNoContentResult(long id)
+    {
+        // Arrange
+        var user = new ClaimsPrincipal(new ClaimsIdentity(
+            new Claim[]
+            {
+                new Claim(ClaimTypes.Role, nameof(Role.TechAdmin).ToLower()),
+            }));
+        controller.ControllerContext.HttpContext.User = user;
+        sensitiveDirectionService.Setup(x => x.Delete(id)).ReturnsAsync(Result<DirectionDto>.Success(direction));
+
+        // Act
+        var result = await controller.DeleteDirectionById(id) as NoContentResult;
+
+        // Assert
+        Assert.That(result, Is.Not.Null);
+        Assert.AreEqual(204, result.StatusCode);
+    }
+
+    [Test]
+    [TestCase(0)]
+    public void DeleteDirectionById_WhenIdIsInvalid_ReturnsBadRequestObjectResult(long id)
```

Comment:
(Naming issue) Test name promises that test `_ReturnsBadRequestObjectResult`, but test code is asserting thrown exception. Please fix

Reply:
Not addressed in last commit